src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -687,7 +687,8 @@ export class TerminalInstance implements ITerminalInstance {
 		this._isExiting = true;
 		this._process = null;
 		let exitCodeMessage: string;
-		if (exitCode) {
+
+		if (exitCode && this._configHelper.config.showExitAlert) {
```

Comment:
Looking at this again I think the `showExitAlert` needs to be added to this line at the bottom of the function:

```js
this._messageService.show(Severity.Error, exitCodeMessage)
```

`exitCodeMessage` is shared between both the message box and also tasks which will be output to the terminal if it fails. In the current state when tasks fail an empty line will be added to the terminal when the task fails instead of the message.

Reply:
Sure! Update.